source/slang/slang-lower-to-ir.cpp

```diff
@@ -2204,6 +2204,16 @@ void addVarDecorations(
             builder->addDecoration(inst, kIROp_GLSLLocationDecoration,
                 builder->getIntValue(builder->getIntType(), stringToInt(glslLocationMod->valToken.getContent())));
         }
+        else if (auto glslInputAttachmentMod = as<GLSLInputAttachmentIndexLayoutModifier>(mod))
+        {
+            auto subpassType = as<IRSubpassInputType>(inst->getDataType());
+
+            if (!subpassType)
+                context->getSink()->diagnose(inst, Diagnostics::InputAttachmentIndexOnlyAllowedOnSubpass);
+
+            builder->addDecoration(inst, kIROp_GLSLInputAttachmentIndexDecoration,
```

Comment:
We should fold the arg expr into a Val and lower that Val here. This will allow users to use an expr instead of just a literal. We need to allow other modifiers to take exprs instead of literals eventually.

Reply:
Should this be done now or in a separate commit where the entirety of `static NodeBase* parseLayoutModifier(Parser* parser, void* /*userData*/)` (and related) is modified to allow expr input?

Currently `GLSLInputAttachmentIndexDecoration` parses through use of the `GLSLLayoutModifier` which most layout attributes resolve from.

Reply:
Let's do it in another PR then.